File: ООП Lab1.3/Lab 1.3/String.cpp
#include "String.h"

CustomString::CustomString() : value("") {}

CustomString::CustomString(const std::string& val) : value(val) {}

CustomString::CustomString(const CustomString& other) : value(other.value) {}

CustomString::~CustomString() {}

int CustomString::length() const {
    return value.length();
}

std::string CustomString::getValue() const {
    return value;
}

CustomString CustomString::operator+(const CustomString& other) const {
    return CustomString(value + other.value);
}

CustomString CustomString::operator*(int times) const {
    std::string result;
    for (int i = 0; i < times; i++) {
        result += value;
    }
    return CustomString(result);
}

File: Lab2.1C#/String.cs
namespace StringLibrary
{
    public class String
    {
        protected string value;

        public String()
        {
            value = "";
        }

        public String(string val)
        {
            value = val;
        }

        public String(String other)
        {
            value = other.value;
        }

        public virtual int Length()
        {
            return value.Length;
        }

        public string GetValue()
        {
            return value;
        }
    }
}